### ISSUE TITLE
`render` should  replace the element

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#1937]
### What version of Hono are you using?

4.0.0-rc.0

### What runtime/platform is your app running on?

any

### What steps can reproduce the bug?

`index.html`:

```html
<!DOCTYPE html>
<html lang="en">
  <body>
    <div id="root"><p>foo</p></div>
    <script type="module" src="/src/main.tsx"></script>
  </body>
</html>
```

`src/main.tsx`:

```tsx
import { render } from 'hono/jsx/dom'

const root = document.getElementById('root') as HTMLElement
render(<p>bar</p>, root)
```

### What is the expected behavior?

<img width="645" alt="Screenshot 2024-01-10 at 16 29 39" src="https://github.com/honojs/hono/assets/10682/bb35ea7e-ec2f-4a7b-9b45-0da37844b844">

### What do you see instead?

<img width="645" alt="Screenshot 2024-01-10 at 16 29 49" src="https://github.com/honojs/hono/assets/10682/c661a435-7c95-4fd7-95bb-489f976bef60">

**Comment by yusukebe** [comment]
Hey @usualoma 

Is this the behavior you expect? I want the `render` to replace the contents. Or I want the hydrate method to replace the content.

**Comment by usualoma** [comment]
@yusukebe 
Thanks for pointing that out.
How about a fix for #1938.

**Comment by yusukebe** [comment]
Fixed by #1938
